File: lib/wasm-micro-runtime-WAMR-1.3.3/samples/spawn-thread/wasm-apps/sum.c
/*
 * Copyright (C) 2019 Intel Corporation.  All rights reserved.
 * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 */

/*
 * have something in bss so that llvm synthesizes
 * wasm start function for this module.
 */
char *
return_bss()
{
    static char bss[4096];
    return bss;
}

int
sum(int start, int length)
{
    int sum = 0, i;

    for (i = start; i < start + length; i++) {
        sum += i;
    }

    return sum;
}

File: lib/wasm-micro-runtime-WAMR-2.4.1/core/iwasm/libraries/wasi-nn/test/models/sum.py
# Copyright (C) 2019 Intel Corporation.  All rights reserved.
# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

import tensorflow as tf

from utils import save_model

model = tf.keras.Sequential([
    tf.keras.layers.InputLayer(input_shape=[5, 5, 1]),
    tf.keras.layers.Conv2D(1, (5, 5), kernel_initializer=tf.keras.initializers.Constant(
        value=1), bias_initializer='zeros'
    )
])

# Export model to tflite

save_model(model, "sum.tflite")
